pkg/types/intoto/v1/intoto.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.types.intoto.v1;
+
+import "google/api/field_behavior.proto";
+import "rekor_common.proto";
+
+option go_package = "github.com/sigstore/rekor/pkg/generated/protobuf/types/intoto/v1";
+option java_package = "dev.sigstore.rekor.types.intoto.v1";
+option java_multiple_files = true;
+option java_outer_classname = "IntotoV1Proto";
+
+/*
+ * ProposedIntoto represents the structure of the message required to present to Rekor
+ * to make an entry of the Intoto type.
+ */
+message ProposedIntoto {
+    /*
+     * DSSE envelope encompassing the entire Intoto statement
```

Comment:
what about reusing the DSSE proto?
https://github.com/secure-systems-lab/dsse/blob/master/envelope.proto

or too much overhead on clients needing to pull that library?